### DISCUSSION TITLE
【读书会】Python自然语言处理

### DISCUSSION THREAD

**Discussion by jiangzhonglian** [apachecn/home#226] [Ideas]
# <center>【读书会】Python自然语言处理<center>

* 【推荐】书籍名称《Python自然语言处理》
* 【推荐】英文地址: <https://www.nltk.org/book>
* 【推荐】翻译地址: <https://usyiyi.github.io/nlp-py-2e-zh>
* 【推荐】API地址: <http://www.nltk.org/py-modindex.html>
* 云链接进度: <https://docs.qq.com/sheet/BDRjaP2oYMYz3FKg3l3FFHWK428LX50jYqxI29AFGo2G46Bv3lmOmZ2cHJf83FYCyH1OggI14#BB08J2>

> 参与

* QQ群： 713436582
* 负责人: 片刻（529815144）  诺木人（498744838）
* 收费：10元 => [**报名地址, 请自己登记信息**](https://docs.qq.com/sheet/BDRjaP2oYMYz3FKg3l3FFHWK428LX50jYqxI29AFGo2G46Bv3lmOmZ2cHJf83FYCyH1OggI14)
* 学完QQ群会解散，解散后大家可以加入社区群
* nlp 想入门，那么就来一起看书学习和讨论碰撞吧！
* 学习: 没有假大空的PPT和无法复显论文，我们喜欢最干的书和代码让你到学习的高潮

> 时间

* 时间周期：2018-08-20～2018-09-02 （共 14 天）
* 每天任务：40页（共382页，预计 12天，共休息2天，双数日子语音直播，单数自己整理笔记）

**[@片刻](https://github.com/jiangzhonglian)** && **[@诺木人](https://github.com/1mrliu)** 会进行语音: 沟通进度和督促学习

> 内容

0. 前言
1. 语言处理与Python
2. 获得文本语料和词汇资源
3. 处理原始文本
4. 编写结构化的程序
5. 分类和词汇标注
6. 学习分类文本
7. 从文本提取信息
8. 分析句子结构
9. 构建基于特征的文法
10. 分析句子的含义
11. 语言学数据管理
12. 后记︰面对语言的挑战

---

历史学习

* 第一期: <https://docs.qq.com/sheet/BDRjaP2oYMYz3KfyAR4N7Y0O10JzK52mZVXO1#BB08J2>

**Comment by xiangking** [comment]
笔记：https://www.jianshu.com/p/dba099750398
目的：学习python怎么处理NLP
完全情况：满意

**Comment by bububut** [comment]
第一天，40页
目的：对中文nlp有兴趣
今天完成：满意，第一天先热热身

**Comment by pisunk** [comment]
目的是：机器学习实战基本看完了，了解机器学习可做的方向大概有nlp和推荐系统，想深入了解下nlp，主要是是nlp框架的使用，尤其是中文nlp框架。
进度：今天看了第一章，主要是一些python基础。

**Comment by jiangzhonglian** [comment]
更新心得：https://apachecn.github.io/home/read/nlp-2-python-nltk
资料：群文件 + 翻译 https://usyiyi.github.io/nlp-py-2e-zh
更新进度：
https://docs.qq.com/sheet/BDRjaP2oYMYz3FKg3l3FFHWK428LX50jYqxI29AFGo2G46Bv3lmOmZ2cHJf83FYCyH1OggI14#BB08J2

2018--8=20 第一天：
第一个问题： 你的目的是什么？
第二个问题：今天完成还满意吗？

**Comment by tabtou** [comment]
目的：互相监督学习。对nlp感兴趣。
效果：看完了第一章python基础

**Comment by zhongleilz** [comment]
第一天：看完了第一章
目的：对nlp感兴趣，
想学习一下nlp的框架
效果：满意

**Comment by xiangking** [comment]
第二天：第二章完成
完成情况：做了笔记，第二章主要还是一些资源的介绍和基础的处理，看下来之后，对处理的对象会有比较大的感悟。
其中WordNet之前没怎么注意，没想到还是知识图谱那一块的知识，倒是有些小意外。

**Comment by xiangking** [comment]
第三天：第三章完成
完成情况：主要还是字符串和正则的知识，多练多写是王道

**Comment by 505555998** [comment]
第三天：完成第三章   
问题：    
特别希望 @片刻 能从负责的项目，给串一串，在什么场景应用到什么模型，里面可能会有哪些坑

**Comment by pisunk** [comment]
第二天：第二章
完成情况：第二章主要是函数较多，记不太熟。
第三天：第三章
完成情况：主要是正则，nltk封装好的分词，分句。
个人也很希望接触一点中文nlp的知识，小项目。

**Comment by jianfengMan** [comment]
相互监督，看书就是快

**Comment by xiangking** [comment]
第四天：第四章完成
主要是python的东西，所以很快就浏览了一遍，有些东西值得多次阅读

**Comment by xiangking** [comment]
第五天：第五章完成
主要是python字典、标注和词性的一些东西

**Comment by jiangzhonglian** [comment]
杭州-幸福秋末凋零  21:59:51
1. 可能在学的过程中没感觉进步很大，但是看完就感觉不一样
2. 感觉学习一拖就容易产生拖延心理
3. 一期一期下来进步很快的（后期一对比就会发现）

---

想起来当年做小白，干kaggle的时候，现在坚持下来，的确天翻地覆！哈哈哈

**Comment by jiangzhonglian** [comment]
最大熵原理是一种选择随机变量统计特性最符合客观情况的准则，也称为最大信息原理。

在投资时常常讲不要把所有的鸡蛋放在一个篮子里，这样可以降低风险。在信息处理中，这个原理同样适用。在数学上，这个原理称为最大熵原理。

**Comment by jiangzhonglian** [comment]
机器学习“判定模型”和“生成模型”有什么区别？

生成模型(generative model)通过学习先验分布来推导后验分布而进行分类，
而判定模型(discriminative model)直接学习后验分布来进行分类。

**Comment by xiangking** [comment]
第六-第九：这几天在外面，没有电脑，所以只能看看书，休息了一天，看完了六、七、八章，这几章开始，NLP的知识扑灭而来，看完之后会对整体的NLP分析和NLP的一些基本任务有比较好的了解和把握。

**Comment by jiangzhonglian** [comment]
8.6 文法开发 -  树库和方法 279 页

NLTK 语料库收集了来自PE08 跨框架跨领域分析器评估共享任务的数据。
一个更大的文法集合已准备好用于比较不同的分析器，它可以通过下载large_grammars 包获得（如：python -m nltk.downloader large_grammars）。
NLTK 语料库也收集了中央研究院树库语料，包括10000 句已分析的句子，来自现代汉语中央研究院平衡语料库。

**Comment by jiangzhonglian** [comment]
英文地址： https://www.nltk.org/book
中文地址： https://usyiyi.github.io/nlp-py-2e-zh
英文API： http://www.nltk.org/py-modindex.html
留言地址： https://apachecn.github.io/home/read/nlp-2-python-nltk

**Comment by jiangzhonglian** [comment]
新手入门项目：

1. 京东文本分类：https://www.jianshu.com/p/28d7d58086a7
    * 代码地址：https://github.com/ChenZhongFu/text-classification-pytorch 
2. nlp-情感分析：https://github.com/apachecn/kaggle/tree/dev/competitions/getting-started/word2vec-nlp-tutorial

---

nlp 相关 GitHub：https://github.com/yanqiangmiffy?tab=repositories

**Comment by xiangking** [comment]
命名实体：https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Sequence-Labeling

**Comment by jiangzhonglian** [comment]
自然语言处理：（推荐一下）
1. https://github.com/liuhuanyong
2. https://liuhuanyong.github.io/

**Comment by xiangking** [comment]
> 自然语言处理：（推荐一下）
> 
> 1. https://github.com/liuhuanyong
> 2. https://liuhuanyong.github.io/

已学习完这里面的命名实体的内容，虽然不是很精致，不过整体没有问题。不过keras封装的太严实，需要自己去找很多资料研究下算法，准备有空做一个pytorch版本的
